### PR TITLE
Adding selectAll keyboard shortcut

### LINKED ISSUE #1099: CTRL+A from selection screen
Usually, the fact that I want to take a screenshot is first in my mind, and then the scope of what the screenshot should contain comes second. So what ends up happening is that I'll hit PrtSc, realize that I want to include my entire screen, and then have to take my hands off the keyboard to grab my mouse. 

It would be nice if using the keyboard shortcut `ctrl+A` would create a selection that covers the entire screen (or moves an existing selection's anchors), for us "screenshot first, edit second" kinds of people. 

Lightshot already has this shortcut, if that's worth noting.

### PR DESCRIPTION
**Pull request by astrosticks**
Closes #1099 by allowing a keyboard shortcut (default `<Ctrl>A`) to select the entire select-able screen. The logic for this mimics the constraints / assumptions in the `moveDirection` and `resizeDirection` methods in `captureWidget.cpp`

### CODE REVIEW

**Review by borgmanJeremy — COMMENTED** [review]
Can someone test this on windows multimonitor before we merge?

File: src/widgets/capture/capturewidget.cpp [hunk 1]
```diff
@@ -941,6 +941,19 @@ void CaptureWidget::downResize()
     }
 }
 
+void CaptureWidget::selectAll()
+{
+    QRect newGeometry = rect();
+    m_selection->setGeometry(newGeometry);
+    m_context.selection = extendedRect(&newGeometry);
+    m_selection->setVisible(true);
+    m_showInitialMsg = false;
```
[the remark this answers is not in the document]
Comment (borgmanJeremy) [reply src/widgets/capture/capturewidget.cpp:950]: Ohh nevermind.

**Comment by borgmanJeremy** [comment]
I think this needs to be adjusted to only maximize on the screen that is currently selected.

**Comment by borgmanJeremy** [comment]
There is a bug on wayland multimonitor. The image below shows the monitor configuration in a VM setup using Fedora33 and wayland. 

image

If the smaller screen 2 is selected before hitting CTRL + A, the toolbars do not appear.

image

**Comment by astrosticks** [comment]
@borgmanJeremy thanks for the review, I'll have to get a similar setup configured since I don't have multiple physical monitors to test with

**Comment by borgmanJeremy** [comment]
I'm on the fence of just merging anyway since I suspect the bug is more related to our wayland issues than the new feature.

Do you need this merged for hacktoberfest or do we have time to investigate more.

**Comment by astrosticks** [comment]
Nope, I've got plenty of time to make this a good first pr! And yeah, based on how I borrowed code from other similar areas to make this feature work, I felt like any issues that arose with this might already have manifested in some other areas.

**Comment by astrosticks** [comment]
@borgmanJeremy can you share with me how you've set up your multi-monitor virtual machine so I can test like that?

**Comment by borgmanJeremy** [comment]
Sure, here are some high level directions and pictures. If you have questions or need more details I should probably write up a guide on this since information was pretty sparse on how to do this. I am running this on a headless server and then streaming to my workstation, but you can do it directly on your workstation.


1. Use Qemu and KVM for virtualization (manage the VM's through virt-manager and use remote-viewer for multihead view)
1. Make sure you have a spice server set up in the config
1. Make sure you have a video QXL driver
1. Make sure you have XML editing enabled and then adjust the following line under VideoQXL:
```

  <model type="qxl" ram="131072" vram="65536" vgamem="65536" heads="2" primary="yes"/>

```

1. Install the VM. After booting into the system make sure the qxl video drivers are installed (many distros ship them by default).
1. Launch remote viewer and connect to your running VM
1.In remote viewer go to View > Display> Display 2

Then you should have 2 displays.


image

**Comment by astrosticks** [comment]
Hey @borgmanJeremy, thanks for the tips. Still having trouble getting kvm to work correctly on my system since I'm not that experienced with virtualization. In the meantime, I was able to test my latest commit (the same one you used) on two real monitors and X, and got the controls to pop up. Do you think there's something tricky about Wayland that's causing your issue? 

I can also look into making Select All only make a selection on the monitor that where the cursor is, if that's still desired. Maybe the best way to do it so that keyboard users can still get a multi-monitor fullscreen screenshot would be to have Select All first make a selection of the monitor with the cursor on it, and then if Select All is the next interaction, then select every monitor.

**Comment by borgmanJeremy** [comment]
@astrosticks I am pretty sure this is a Wayland issue and not an issue with the new feature.

I am okay either way if we select 1 or all monitors. The only thing I would like to see complete is this hotkey added into the new hotkey system.

**Pull request merged**